NuRadioReco/modules/channelGenericNoiseAdder.py

```diff
@@ -353,11 +413,31 @@ def bandlimited_noise_from_spectrum(self, n_samples, sampling_rate, spectrum, am
         elif type == 'rayleigh':
             fsigma = amplitude * sigscale / np.sqrt(2.)
             ampl[selection] = self.__random_generator.rayleigh(fsigma, n_samples_freq)
+        elif type == "data-driven":
+            if station_id is None or channel_id is None:
+                self.logger.error("When selecting data-driven noise, the station and channel ids should be passed to bandlimeted noise")
+                raise ValueError
+            
+            scale_parameter_path = f"thermal_noise_scale_parameters_s{station_id}_season23.json"
+            if scale_parameter_path in self.scale_parameter_paths:
+                scale_parameter_full_path = self.scale_parameter_dir + "/" + scale_parameter_path
+            else:
+                raise NotImplementedError("Other station parameters are being generated")
+            
+            scale_parameters = load_scale_parameters(scale_parameter_full_path)
+            fsigma = scale_parameters[channel_id](frequencies[selection])
+            ampl[selection] = self.__random_generator.rayleigh(fsigma, n_samples_freq)
```

Comment:
since this code is a copy from the one in the other function you could consider putting this into its own function.